### ISSUE TITLE
Package spatie/data-transfer-object is abandoned, switch to spatie/laravel-data

### ISSUE THREAD

**Issue by MarioVillani** [knuckleswtf/scribe#589]
Hello,
i get the following warning when installing the package via composer:

`Package spatie/data-transfer-object is abandoned, you should avoid using it. Use spatie/laravel-data instead.`

Is it hard to switch to the new package?
Doing a quick code research,  i see that impacted classes are in **/camel** folder:

    BaseDTO.php
    BaseDTOCollection.php`

Thanks in advance!

**Comment by repat** [comment]
@MarioVillani See https://github.com/knuckleswtf/scribe/issues/546#issuecomment-1302836669. When looking for issue, make sure to include closed ones ;) 

I'm currently looking into replacing `spatie/data-transfer-object` with `spatie/laravel-data` and creating a PR, as @shalvah said "sure, give it a shot". I'll let you know what I find.

**Comment by bram-pkg** [comment]
@repat how's that going? :)

**Comment by tomcoonen** [comment]
@shalvah You closed this issue but the dependency is not replaced? Anything we can do to help?

**Comment by shalvah** [comment]
Dudeeee...did you not see my response, which has already been linked to (_and_ quoted) in this thread? My answer has not changed.

Guys, if you want the package replaced, then do the work. I have too many other things going on in my life to spend hours figuring out how to migrate to a package that has zero benefit for me OR this package. Ffs, the ONLY benefit to you as a user is to make your console a little less noisy. That's hours of unpaid, unfulfilling work. What gives?

I already spend enough (unpaid) time addressing issues from users every week, and working on the actual functionality of this package. Would you prefer I spend my limited time on this? Fuck, even if you sent in a PR, I'd still have to do the work of double-checking it.

In conclusion: this is the lowest of all low priorities imaginable. Heck, it's not even a priority. Even if you sent a PR, I can't guarantee I'd merge it in any time, because it's still work for me (reviewing and maintaining afterwards). So please don't ask me about this again.

**Comment by tomcoonen** [comment]
Thanks for the response, I somehow did not see the comment or interpreted it correctly since the issue was closed.
I agree with you, just seeing if we can help to get it replaced..

@repat Did you make any progress?

**Comment by shalvah** [comment]
Good, thank you for understanding. Closed the issue because when I look at open issues, I want to see things that need my attention. Muting it as well. Adios.👋

**Comment by repat** [comment]
> Thanks for the response, I somehow did not see the comment or interpreted it correctly since the issue was closed. I agree with you, just seeing if we can help to get it replaced..
> 
> @repat Did you make any progress?

@tomcoonen sry I just didn't have the time yet

**Comment by codespearhead** [comment]
@jhoanborges Are you willing to submit a PR?

Because I currently don't have time to tackle it myself, but I can help if you get stuck.

Friendly reminder that open-source projects rely heavily on the community to stay viable, and by that I'm not necessarily mean financial contributions: triaging issues and submitting patches go a long way.

**Comment by m-ah07** [comment]
Hi,  
I recently installed `spatie/laravel-data` as a replacement for `spatie/data-transfer-object`, based on the deprecation notice:  

> *Package spatie/data-transfer-object is abandoned, you should avoid using it. Use spatie/laravel-data instead.*  

However, I am facing issues when trying to remove `spatie/data-transfer-object`. When running `composer remove spatie/data-transfer-object`, it seems to cause conflicts or dependencies not resolving properly.  

Has anyone else encountered this issue? What would be the recommended approach to smoothly transition from `spatie/data-transfer-object` to `spatie/laravel-data` without breaking existing functionality?  

Any guidance or best practices would be highly appreciated.

**Comment by bram-pkg** [comment]
You can't remove `spatie/data-transfer-object` from your project because Scribe requires it. You would need to PR a refactor to Scribe to replace the package.

**Comment by Vectorial1024** [comment]
Would #1024 be a good candidate to close this?

@shalvah I understand what you mean by having to rat-race packages; I also have a few open source packages myself. But, if you are looking for contributions, then #1024 might be what you need.

**Comment by shalvah** [comment]
Checking...

**Comment by Norda-AD** [comment]
As this was addressed in the latest release, it might be a good time to unpin it and reduce clutter.
